### PR TITLE
[Lua] Fix several Dalham (CoP 7-4 mob) issues

### PR DESCRIPTION
**Pull request by TracentEden2**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
This PR fixes several issues with Dalham:
1. Add 100% enwater additional effect
2. Add the correct immunities, resistances, and water absorb
3. Add the correct increase in attacks per round and the 2hr animation at different HP levels
4. Change to the correct war job, correct HP, and correct delay

The changes are based on the captures by Siknoz [here](https://youtu.be/u9APAn_F5TU), zach2good [here](https://drive.google.com/open?id=157JfVoVzba4f_4DV1-q9GYV0pewuFkpH), and Jimmayus [here](https://youtu.be/0WmmSZvB77g).

## Steps to test these changes
Fight Dalham and compare to retail capture videos and information

### CODE REVIEW

**Review by zach2good — APPROVED** [review]

File: scripts/zones/Bibiki_Bay/mobs/Dalham.lua [hunk 1]
```diff
@@ -7,13 +7,44 @@ local entity = {}
...
-    -- "Increases swings per attack round as its HP lowers. At 15-20% swings like Charybdis giving the illusion of Hundred Fists."
-    local swings = 1 + math.floor((100 - mob:getHPP()) / 18)
-    mob:setMobMod(xi.mobMod.MULTI_HIT, swings)
+    if mob:getHPP() < 75 and mob:getLocalVar('multihitControl') == 0 then
+        mob:setLocalVar('multihitControl', 1)
+        -- set DOUBLE_ATTACK to 100
+        -- cannot use MULTI_HIT because that is just upto X hits
+        mob:setMod(xi.mod.DOUBLE_ATTACK, 100)
+        -- dummy mob ability for 2hr animation (does not use tp)
+        mob:useMobAbility(626)
```
Comment (WinterSolstice8) [inline scripts/zones/Bibiki_Bay/mobs/Dalham.lua]: Based on the cap there's no way this is actually a real ability and you should just use injectActionPacket

`mob:injectActionPacket(mob:getID(), 11, 437, 0, 0x18, 0, 0, 626)`
most 2hr dustclouds like this should probably just be injections

(note the back to back WS/finish packets)
image

**Review by WinterSolstice8 — CHANGES_REQUESTED** [review]

File: scripts/zones/Bibiki_Bay/mobs/Dalham.lua (hunk 1 above)
Comment (TracentEden2) [reply scripts/zones/Bibiki_Bay/mobs/Dalham.lua]: > Based on the cap there's no way this is actually a real ability and you should just use injectActionPacket
> 
> `mob:injectActionPacket(mob:getID(), 11, 437, 0, 0x18, 0, 0, 626)` most 2hr dustclouds like this should probably just be injections

Thanks for the info, I have updated the PR to use `injectActionPacket` instead!

**Review by WinterSolstice8 — APPROVED** [review]

File: scripts/zones/Bibiki_Bay/mobs/Dalham.lua [hunk 2]
```diff
@@ -7,13 +7,44 @@ local entity = {}
 
 entity.onMobInitialize = function(mob)
     mob:setMobMod(xi.mobMod.IDLE_DESPAWN, 180)
+end
+
+entity.onMobSpawn = function(mob)
+    mob:setMobMod(xi.mobMod.ADD_EFFECT, 1)
+    mob:setMod(xi.mod.SLEEPRES, 90)
```
Comment (Xaver-DaRed) [inline scripts/zones/Bibiki_Bay/mobs/Dalham.lua]: This modifier is exclusively used for job traits and similar, which make it so it returns an special resist message, allowing us to know it was the job trait that triggered

There should be a MEVA mod for sleep

File: scripts/zones/Bibiki_Bay/mobs/Dalham.lua [hunk 3]
```diff
@@ -7,13 +7,44 @@ local entity = {}
...
+        -- inject packet for 2hr animation
+        mob:injectActionPacket(mob:getID(), 11, 437, 0, 0x18, 0, 0, 626)
+    elseif mob:getHPP() < 25 and mob:getLocalVar('multihitControl') == 2 then
+        mob:setLocalVar('multihitControl', 3)
+        -- set QUAD_ATTACK to 100
+        mob:setMod(xi.mod.TRIPLE_ATTACK, 0)
+        mob:setMod(xi.mod.QUAD_ATTACK, 100)
+        -- inject packet for 2hr animation
+        mob:injectActionPacket(mob:getID(), 11, 437, 0, 0x18, 0, 0, 626)
+    end
```
Comment (Xaver-DaRed) [inline scripts/zones/Bibiki_Bay/mobs/Dalham.lua:50]: We could store both the mob_local_variable and the mob_HPP into local variables before the if/elseif chain, so we dont call them potentially 3 times per tick.

File: scripts/zones/Bibiki_Bay/mobs/Dalham.lua (hunk 2 above)
Comment (TracentEden2) [reply scripts/zones/Bibiki_Bay/mobs/Dalham.lua]: > This modifier is exclusively used for job traits and similar, which make it so it returns an special resist message, allowing us to know it was the job trait that triggered
> 
> There should be a MEVA mod for sleep

Thanks, I have updated the PR!

File: scripts/zones/Bibiki_Bay/mobs/Dalham.lua (hunk 3 above)
Comment (TracentEden2) [reply scripts/zones/Bibiki_Bay/mobs/Dalham.lua:50]: > We could store both the mob_local_variable and the mob_HPP into local variables before the if/elseif chain, so we dont call them potentially 3 times per tick.

Thanks, I have updated the PR!

**Review by Xaver-DaRed — APPROVED** [review]

**Pull request merged**